includes/Checker/Checks/Plugin_Repo/File_Type_Check.php

```diff
@@ -474,6 +478,147 @@ protected function look_for_composer_files( Check_Result $result, array $files )
+					$found_ai_dirs[ $ai_dir ] = true;
+					break;
+				}
+			}
+		}
+
+		foreach ( array_keys( $found_ai_dirs ) as $ai_dir ) {
+			$this->add_result_message_for_file(
+				$result,
+				$is_error,
+				sprintf(
+					/* translators: %s: directory name */
+					__( 'AI instruction directory "%s" detected. These directories should not be included in production plugins.', 'plugin-check' ),
+					$ai_dir
+				),
+				'ai_instruction_directory',
+				$plugin_path . $ai_dir,
+				0,
+				0,
+				'',
+				8
+			);
+		}
+	}
+
+	/**
+	 * Checks for GitHub workflow directory.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param Check_Result $result   Check result to amend.
+	 * @param array        $files    List of file paths.
+	 * @param bool         $is_error Whether to report as error.
+	 */
+	protected function check_github_directory( Check_Result $result, array $files, $is_error ) {
+		$plugin_path  = $result->plugin()->path();
+		$found_github = false;
+
+		foreach ( $files as $file ) {
+			$relative_path = str_replace( $plugin_path, '', $file );
+			if ( strpos( $relative_path, '/.github/' ) !== false || strpos( $relative_path, '.github/' ) === 0 ) {
+				$found_github = true;
+				break;
+			}
+		}
+
+		if ( $found_github ) {
+			$this->add_result_message_for_file(
+				$result,
+				$is_error,
+				__( 'GitHub workflow directory ".github" detected. This directory should not be included in production plugins.', 'plugin-check' ),
+				'github_directory',
+				$plugin_path . '.github',
+				0,
+				0,
+				'',
+				8
+			);
+		}
+	}
+
+	/**
+	 * Checks for unexpected markdown files.
+	 *
+	 * @since 1.8.0
+	 *
+	 * @param Check_Result $result   Check result to amend.
+	 * @param array        $files    List of file paths.
+	 * @param bool         $is_error Whether to report as error.
+	 */
+	protected function check_unexpected_markdown_files( Check_Result $result, array $files, $is_error ) {
+		$plugin_path           = $result->plugin()->path();
+		$allowed_root_md_files = array( 'README.md', 'readme.txt', 'LICENSE', 'LICENSE.md', 'CHANGELOG.md', 'CONTRIBUTING.md' );
+		$root_md_files         = array();
+
+		foreach ( $files as $file ) {
+			$relative_path = str_replace( $plugin_path, '', $file );
+			$relative_path = ltrim( $relative_path, '/' );
+			$basename      = basename( $file );
+
+			if ( substr_count( $relative_path, '/' ) === 0 && pathinfo( $file, PATHINFO_EXTENSION ) === 'md' ) {
+				if ( ! in_array( $basename, $allowed_root_md_files, true ) ) {
+					$root_md_files[] = $file;
+				}
+			}
+		}
+
+		foreach ( $root_md_files as $file ) {
+			$this->add_result_message_for_file(
+				$result,
+				$is_error,
+				sprintf(
+					/* translators: %s: file name */
+					__( 'Unexpected markdown file "%s" detected in plugin root. Only README.md, readme.txt, LICENSE, LICENSE.md, CHANGELOG.md, and CONTRIBUTING.md are expected in production plugins.', 'plugin-check' ),
```

Comment:
Lets avoid list of files here in the message as this list would changing timely.